tests/components/efergy/test_sensor.py

```diff
@@ -67,11 +82,31 @@ async def test_single_sensor_readings(
     assert await async_setup_component(hass, "sensor", {"sensor": ONE_SENSOR_CONFIG})
     await hass.async_block_till_done()
 
-    assert hass.states.get("sensor.energy_consumed").state == "38.21"
-    assert hass.states.get("sensor.energy_usage").state == "1580"
-    assert hass.states.get("sensor.energy_budget").state == "ok"
-    assert hass.states.get("sensor.energy_cost").state == "5.27"
-    assert hass.states.get("sensor.efergy_728386").state == "1628"
+    state = hass.states.get("sensor.energy_usage")
+    assert state.state == "1580"
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == DEVICE_CLASS_POWER
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == POWER_WATT
+    assert state.attributes.get(ATTR_STATE_CLASS) == STATE_CLASS_MEASUREMENT
+    state = hass.states.get("sensor.energy_consumed")
+    assert state.state == "38.21"
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == DEVICE_CLASS_ENERGY
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == ENERGY_KILO_WATT_HOUR
+    assert state.attributes.get(ATTR_STATE_CLASS) == STATE_CLASS_TOTAL_INCREASING
+    state = hass.states.get("sensor.energy_budget")
+    assert state.state == "ok"
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == DEVICE_CLASS_MONETARY
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == CURRENCY_DOLLAR
+    assert state.attributes.get(ATTR_STATE_CLASS) is None
+    state = hass.states.get("sensor.energy_cost")
+    assert state.state == "5.27"
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == DEVICE_CLASS_MONETARY
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == "$/day"
```

Comment:
The monetary device class sensor must only have an ISO-standard currency code as unit. Not a per day unit.